clang/lib/Driver/Driver.cpp

```diff
@@ -5399,14 +5416,26 @@ InputInfo Driver::BuildJobsForActionNoCache(
                                                 DependentOffloadKind)}] =
           CurI;
     }
-
-    // Now that we have all the results generated, select the one that should be
-    // returned for the current depending action.
-    std::pair<const Action *, std::string> ActionTC = {
-        A, GetTriplePlusArchString(TC, BoundArch, TargetDeviceOffloadKind)};
-    assert(CachedResults.find(ActionTC) != CachedResults.end() &&
-           "Result does not exist??");
-    Result = CachedResults[ActionTC];
+    // Do a check for a dependency file unbundle for FPGA.  This is out of line
+    // from a regular unbundle, so just create and return the name of the
+    // unbundled file.
+    if (JA->getType() == types::TY_FPGA_Dependencies) {
+      // Unbundling of dependency file for FPGA is target only
+      std::string TmpFileName = C.getDriver().GetTemporaryPath(
+            llvm::sys::path::stem(BaseInput), "d");
+      const char *TmpFile =
+          C.addTempFile(C.getArgs().MakeArgString(TmpFileName));
+      Result = InputInfo(types::TY_Dependencies, TmpFile, TmpFile);
+      UnbundlingResults.push_back(Result);
```

Comment:
Why don't we have to explicitly create a tempfile for most of the actions's outputs, but require adding this construction here?

Reply:
The generation of the tempfiles are done for unbundling of particular cases above this area (see `CurI` assignments) but in general are created in `Driver::GetNamedOutputPath` based on prefix, JobAction, etc.  As this is not a typical unbundle step, no dep information is created when the unbundling Job is added so we create the temp file explicitly.

Reply:
Got it, thanks for the pointers!